extensions/wikia/FounderEmails/events/FounderEmailsEditEvent.class.php

```diff
@@ -43,151 +147,121 @@ public function enabled ( $wgCityId, User $user ) {
-				wfProfileOut( __METHOD__ );
-				return true;
-			}
-
-			$foundingWiki = WikiFactory::getWikiById( $wgCityId );
-			$founderEmailObj = FounderEmails::getInstance();
-			$wikiService = ( new WikiService );
-			$user_ids = $wikiService->getWikiAdminIds();
-			$emailParams = array(
-				'$EDITORNAME' => $eventData['data']['editorName'],
-				'$EDITORPAGEURL' => $eventData['data']['editorPageUrl'],
-				'$EDITORTALKPAGEURL' => $eventData['data']['editorTalkPageUrl'],
-				'$MYHOMEURL' => $eventData['data']['myHomeUrl'],
-				'$WIKINAME' => $foundingWiki->city_title,
-				'$PAGETITLE' => $eventData['data']['titleText'],
-				'$PAGEURL' => $eventData['data']['titleUrl'],
-				'$WIKIURL' => $foundingWiki->city_url,
-			);
+	private function hasExceededNotificationLimit( User $user ) {
+		$count = $this->getUserNotificationCount( $user );
+		return $count > self::DAILY_NOTIFICATION_LIMIT;
+	}
 
-			$msgKeys = array();
-			$today = date( 'Y-m-d' );
-			$wikiType = !empty( $wgEnableAnswers ) ? '-answers' : '';
+	private function hasHitNotificationLimit( User $user ) {
+		$count = $this->getUserNotificationCount( $user );
+		return $count == self::DAILY_NOTIFICATION_LIMIT;
+	}
 
-			foreach ( $user_ids as $user_id ) {
-				$user = User::newFromId( $user_id );
+	private function getUserNotificationCount( User $user ) {
+		$counter = $this->getNotificationCounter( $user );
+		return $counter[F::app()->wg->CityId]['count'];
+	}
 
-				// skip if not enable
-				if ( !$this->enabled( $wgCityId, $user ) ) {
-					continue;
-				}
+	/**
+	 * Keeps a per request cache of the 'founderemails-counter' user option
+	 *
+	 * @param User $user
+	 *
+	 * @return mixed
+	 */
+	private function getNotificationCounter( User $user ) {
+		static $counter = null;
+
+		if ( empty( $counter[$user->getId()] ) ) {
+			$allCounter = unserialize( $user->getOption( 'founderemails-counter' ) );
+			if ( $this->userCounterNeedsReset( $allCounter ) ) {
+				$allCounter[F::app()->wg->CityId] = [
+					'date'  => date( 'Y-m-d' ),
+					'count' => 0,
+				];
+			}
 
-				// skip if reciever is the editor
-				if ( $user->getName() == $eventData['data']['editorName'] ) {
-					continue;
-				}
+			$counter[$user->getId()] = $allCounter;
+		}
 
-				// BugID: 1961 Quit if the founder email is not confirmed
-				if ( !$user->isEmailConfirmed() ) {
-					wfProfileOut( __METHOD__ );
-					return true;
-				}
+		return $counter[$user->getId()];
+	}
 
-				$aAllCounter = unserialize( $user->getOption( 'founderemails-counter' ) );
-				if ( empty( $aAllCounter ) ) {
-					$aAllCounter = array();
-				}
+	private function userCounterNeedsReset( array $counter ) {
+		if ( empty( $counter[F::app()->wg->CityId] ) ) {
+			return true;
+		}
 
-				// quit if the Founder has recieved enough emails today
+		$today = date( 'Y-m-d' );
+		$wikiCounter = $counter[F::app()->wg->CityId];
+		return empty( $wikiCounter['date'] ) || $wikiCounter['date'] !== $today;
```

Comment:
I'm glad you trust my initialization code above!  I've been slapped by PHP when I haven't been so rigorous, but I suppose we should always have 'date' defined here...